src/components/Questionnaire/utils.ts

```diff
@@ -0,0 +1,81 @@
+import { Question } from "@/types/questionnaire/question";
+
+export const removeQuestionsFromSource = (
+  questions: Question[],
+  selectedQuestionIds: Set<string>,
+): Question[] => {
+  return questions.map((q) => ({
+    ...q,
+    questions: q.questions
+      ? q.questions
+          .filter((subQ) => !selectedQuestionIds.has(subQ.id))
+          .map((subQ) => ({
+            ...subQ,
+            questions: subQ.questions
+              ? removeQuestionsFromSource(subQ.questions, selectedQuestionIds)
+              : subQ.questions,
+          }))
+      : q.questions,
+  }));
+};
+
+export const addQuestionsToDestination = (
+  questions: Question[],
+  destId: string,
+  questionsToAdd: Question[],
+): Question[] => {
+  return questions.map((q) => {
+    if (q.id === destId) {
```

Comment:
only one qn is going to match the destination Id, but since map is being used here, it's gonna go through the all the questions passed as a param.

removeQuestionsFromSource can also be optimized 🤔